commands/shooter/aimsystem.py

```diff
@@ -0,0 +1,106 @@
+        self.shoot.setWheelSpeed(wheelSpeed)
+        self.shoot.setHoodAngle(Rotation2d.fromDegrees(hoodAngle))
+        self.shoot.rotateTurret(
+            Rotation2d.fromDegrees(turretPosition)
+            + constants.kOffsetAngleRange * self.offset.sideToSide()
+        )
+
+    def normalMode(self) -> None:
+        deltaPos = Pose2d(
+            *[
+                constants.kRobotUpdateRate * i * constants.kPredictiveAimGain
+                for i in SmartDashboard.getNumberArray(
+                    constants.kDriveVelocityKeys, [0, 0, 0]
+                )
+            ]
+        )
+        currentPose = Pose2d(
+            *SmartDashboard.getNumberArray(
+                constants.kRobotPoseArrayKeys.valueKey, [0, 0, 0]
+            )
+        )
+
+        staticDifference = Transform2d(currentPose, constants.kSimDefaultTargetLocation)
+        staticRotation = rotationFromTranslation(staticDifference.translation())
+
+        newPosition = Pose2d(
+            deltaPos.X() + currentPose.X(),
+            deltaPos.Y() + currentPose.Y(),
+            deltaPos.rotation() + currentPose.rotation(),
+        )
+
+        movingDifference = Transform2d(newPosition, constants.kSimDefaultTargetLocation)
+        movingRotation = rotationFromTranslation(movingDifference.translation())
+
+        deltaRotation = movingRotation - staticRotation
+
+        if SmartDashboard.getBoolean(
+            constants.kTargetDistanceRelativeToRobotKeys.validKey, False
+        ):
+            distance = (
+                SmartDashboard.getNumber(
+                    constants.kTargetDistanceRelativeToRobotKeys.valueKey, 0.0
+                )
+                + constants.kOffsetDistanceRange * self.offset.forwardsBackwards()
+            )
+
+            targetSpeed = constants.kShootingMappingFunction(distance)
+            self.shoot.setWheelSpeed(targetSpeed)
+
+            targetAngle = Rotation2d.fromDegrees(
+                constants.kHoodMappingFunction(distance)
+            )
+            self.shoot.setHoodAngle(targetAngle)
+
+        if SmartDashboard.getBoolean(
+            constants.kTargetAngleRelativeToRobotKeys.validKey, False
+        ):
+            angle = (
+                SmartDashboard.getNumber(
+                    constants.kTargetAngleRelativeToRobotKeys.valueKey, 0.0
+                )
+                + constants.kOffsetAngleRange.degrees() * self.offset.sideToSide()
+            )
+
+            self.shoot.trackTurret(
+                angle + deltaRotation.degrees()
+            )  # always track the turret
+        else:
+            self.shoot.rotateTurret(
+                optimizeAngle(
+                    constants.kTurretForwardAngle,
+                    movingRotation
+                    + constants.kOffsetAngleRange * self.offset.sideToSide(),
+                )
+                + Rotation2d.fromDegrees(180)  # shooter 0 is robot 180
```

Comment:
Would be good to have a comment describing why different behaviors are necessary depending on whether the target angle is valid/invalid and what the intended behavior of the aiming system should be in each case